src/core/diags/factory_diags.cpp

```diff
@@ -550,7 +551,14 @@ void Diags::TransmitPacket(void)
     }
 
     mDiagSendOn = true;
-    IgnoreError(Get<Radio>().Transmit(*static_cast<Mac::TxFrame *>(mTxPacket)));
+    error = Get<Radio>().Transmit(*static_cast<Mac::TxFrame *>(mTxPacket));
+
+    if(error != kErrorNone)
+    {
+        Output("Packet transmission failed, error: %s\r\n", ErrorToString(error));
+    }
```

Comment:
I have submitted a PR https://github.com/openthread/openthread/pull/11108 to record the number of packets that are sent failed. Could you consider adding a new variable `mSentErrorInvalidState` to `mStats` to records the failure rather than outputting the error message? Because running `diag repeat` in the invalid state may output endless error messages.

Reply:
sure! aligned